spynnaker/pyNN/models/neuron/population_vertex.py

```diff
@@ -543,13 +727,21 @@ def add_incoming_projection(self, projection: Projection) -> None:
         # Reset the ring buffer shifts as a projection has been added
         SpynnakerDataView.set_requires_mapping()
         self.__max_row_info.clear()
+        self.__max_delay_ms = None
+        self.__max_delay_slots_available = None
+        self.__n_synapse_cores = self.__n_synapse_cores_param
         # pylint: disable=protected-access
         pre_vertex = projection._projection_edge.pre_vertex
         self.__incoming_projections[pre_vertex].append(projection)
         if pre_vertex == self:
             self.__self_projection = projection
         if isinstance(pre_vertex, SpikeSourcePoissonVertex):
-            self.__incoming_poisson_projections.append(projection)
+            # Only keep these when there is only one projection to this
+            # vertex, as otherwise not interesting
+            if pre_vertex not in self.__incoming_poisson_projections:
```

Comment:
is looks dangerous!

the first time add the Projection
the second time del the Projection
the third time will add it again!